bgpd/bgp_zebra.c

```diff
@@ -1435,15 +1435,25 @@ void bgp_zebra_announce(struct bgp_node *rn, struct prefix *p,
 		for (i = 0; i < api.nexthop_num; i++) {
 			api_nh = &api.nexthops[i];
 
-			if (api_nh->type == NEXTHOP_TYPE_IFINDEX)
+			switch (api_nh->type) {
+			case NEXTHOP_TYPE_IFINDEX:
 				nh_buf[0] = '\0';
-			else {
-				if (api_nh->type == NEXTHOP_TYPE_IPV4)
-					nh_family = AF_INET;
-				else
-					nh_family = AF_INET6;
+				break;
+			case NEXTHOP_TYPE_IPV4:
+			case NEXTHOP_TYPE_IPV4_IFINDEX:
+				nh_family = AF_INET;
 				inet_ntop(nh_family, &api_nh->gate, nh_buf,
 					  sizeof(nh_buf));
+				break;
+			case NEXTHOP_TYPE_IPV6:
+			case NEXTHOP_TYPE_IPV6_IFINDEX:
+				nh_family = AF_INET6;
+				inet_ntop(nh_family, &api_nh->gate, nh_buf,
+					  sizeof(nh_buf));
+				break;
+			case NEXTHOP_TYPE_BLACKHOLE:
+				strlcpy(nh_buf, "blackhole", sizeof(nh_buf));
+				break;
```

Comment:
The code looks good. Having a default case with an error log in this switch would help for better debugging. Otherwise, this would be a silent failure in case api_nh->type has some garbage value or unsupported value.

Let me know your thoughts, I will approve accordingly.

Reply:
No!  Adding a default would kill the ability to know where we need to make changes to code when the enum changes.  All cases of the enum are handled.  We'll quickly crash with garbage values of nexthop type

Reply:
Agree to some extent. But, unless there is an assert for unknown cases, not having default case also will not help in identifying the new code changes required. Isn't it ?